resources/assets/coffee/react/_components/beatmapset-panel.coffee

```diff
@@ -50,16 +50,27 @@ class @BeatmapsetPanel extends React.PureComponent
     # arbitrary number
     maxDisplayedDifficulty = 10
 
+    condenseDifficulties = beatmapset.beatmaps.length > maxDisplayedDifficulty
+
+    beatmapIcon = (beatmap) ->
+      div
+        className: "#{'beatmapset-panel__difficulty-icon' if !condenseDifficulties}"
+        key: beatmap.id
+        el BeatmapIcon, beatmap: beatmap
+
     difficulties =
-      for own _mode, beatmaps of BeatmapHelper.group beatmapset.beatmaps[..maxDisplayedDifficulty - 1]
-        for b in beatmaps
-          div
-            className: 'beatmapset-panel__difficulty-icon'
-            key: b.id
-            el BeatmapIcon, beatmap: b
-
-    if beatmapset.beatmaps.length > maxDisplayedDifficulty
-      difficulties.push span key: 'over', "+#{(beatmapset.beatmaps.length - maxDisplayedDifficulty)}"
+      for own _mode, beatmaps of BeatmapHelper.group beatmapset.beatmaps
+        if condenseDifficulties
+          [..., hardestDiff] = beatmaps
+          [ 
```

Comment:
can also use `span` here and use mode as `key` so the containing icon and count don't need key.